src/NuGet.Core/NuGet.Commands/RestoreCommand/Diagnostics/UnexpectedDependencyMessages.cs

```diff
@@ -225,9 +224,23 @@ public static bool HasMissingLowerBound(VersionRange range)
         }
 
         /// <summary>
-        /// Log upgrade warnings from the graphs.
+        /// Logs upgrade warnings from the graphs.
         /// </summary>
+        /// <param name="graphs">Flattened restore graphs with error and warning info</param>
+        /// <param name="logger">Not used</param>
+        /// <returns>A collection of log messages</returns>
+        [Obsolete("Use GetDependenciesAboveUpperBounds(List<IndexedRestoreTargetGraph>) instead")]
         public static IEnumerable<RestoreLogMessage> GetDependenciesAboveUpperBounds(List<IndexedRestoreTargetGraph> graphs, ILogger logger)
```

Comment:
Given that these are restore internals and this is a major version change, i think we should *break* the API rather than add more overloads.

Reply:
Removed